app/src/main/java/com/github/h3lp3rs/h3lp/forum/ForumCategory.kt

```diff
@@ -23,17 +29,31 @@ enum class ForumCategory {
         fun forumOf(choice: ForumCategory): Forum {
             // The categories are the first sub-level of the forum, thus they lie after root (the
             // empty list)
-            choice.forum = choice.forum ?: FireForum(listOf(choice.name))
-            return choice.forum!!
+            root?.let {
+                return it.child(choice.name)
+            }
+            root = FireDBForum(emptyList(), FireDatabase(ROOT_FORUM_DB_PATH))
+            return root!!.child(choice.name)

```

Comment:
an alternative to avoid the !! operator since it's usually bad practice:
```suggestion
            val forum = FireDBForum(emptyList(), FireDatabase(ROOT_FORUM_DB_PATH))
            root = forum
            return forum.child(choice.name)
```

Reply:
I want to push this, but I won't waste a whole Cirrus build for this.